crates/ruff/src/rules/pycodestyle/mod.rs

```diff
@@ -53,6 +54,26 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn w292_4() -> Result<()> {
+        let diagnostics = test_path(
+            Path::new("pycodestyle/W292_4.py"),
+            &settings::Settings::for_rule(Rule::NoNewLineAtEndOfFile),
+        )?;
+
+        assert_yaml_snapshot!(
+            diagnostics,
+            // Replaces the platform's default line ending with `<LineEnding>` to make the test platform-
+            // agnostic
+            {
+                "[].fix.content" => insta::dynamic_redaction(|value, _path| {
+                    value.as_str().unwrap().replace(LineEnding::default().as_str(), "<LineEnding>")
+                })
```

Comment:
This is the only test where the output depends on where the test is run. That's why I'm handling this test specifically.

Reply:
In https://github.com/charliermarsh/ruff/pull/2033 we chose not to use a special line ending token. In this case however, since it's only the approach taken there would not much differ. (cc \andersk who came up with the idea to split newlines).